### PR TITLE
[github actions] Fix token issue on actions/checkout package

### PR DESCRIPTION
**Pull request by XilaiZhang**
revision 01/17: instead of removing actions/checkout, keep actions/checkout but remove the `token` field and add `persist-credentials` field. tested with a [mirror script](https://github.com/XilaiZhang/miscellaneous-side-project/blob/master/.github/workflows/easy-cp.yml) and creates [expected pull request](https://github.com/flutter/flutter/pull/141730)

Issue: when running github actions, the [tokens not found error](https://github.com/actions/checkout/issues/298) still happens( `Input required and not supplied: token`). We are not using fork PR or dependabot, and it's flaky when the well defined token isn't find in the inputs. We hit this error when invoking [market place actions/checkout](https://github.com/actions/checkout): [example failed run 1](https://github.com/flutter/flutter/actions/runs/7546108771/job/20543199801), [example failed run 2](https://github.com/flutter/flutter/actions/runs/7546141972/job/20543265842)

In this PR, Remove the dependency on marketplace actions  to make our workflow more reliable and less flaky.

other changes to remove actions/checkout dependency:
1. embedded token url for git push
Tried a number of ways and this is the only / best workaround I found to resolve [the notorious problem of pushing without ssh key](https://stackoverflow.com/questions/22147574/github-fatal-could-not-read-username-for-https-github-com-no-such-file-o). 
2. added back `--head`
`--head` is now needed to avoid [abort](https://github.com/XilaiZhang/miscellaneous-side-project/actions/runs/7548409479/job/20550397014)

Test: a [replicate of the actions file](https://github.com/XilaiZhang/miscellaneous-side-project/blob/master/.github/workflows/easy-cp.yml) is [tested](https://github.com/XilaiZhang/miscellaneous-side-project/actions/runs/7548448024/job/20550521341) on my personal repo, and it creates the [expected PR](https://github.com/flutter/flutter/pull/141647)

### CODE REVIEW

File: .github/workflows/easy-cp.yml [hunk 1]
```diff
@@ -58,8 +52,8 @@ jobs:
         if: ${{ steps.attempt-cp.conclusion == 'success' }}
         working-directory: ./flutter
         run: |
-          git push -u origin cp-${CHANNEL}-${COMMIT_SHA}
-          gh pr create --title "[CP-${CHANNEL}]${PR_TITLE}" --body-file ../PULL_REQUEST_CP_TEMPLATE.md --base ${RELEASE_BRANCH} --label "cp: review" --repo flutter/flutter
+          git push https://${GITHUB_TOKEN}@github.com/flutteractionsbot/flutter cp-${CHANNEL}-${COMMIT_SHA}
```
Comment (CaseyHillers) [inline .github/workflows/easy-cp.yml:55]: Does `${GITHUB_TOKEN}@github.com` leak the secret in the logs?

File: .github/workflows/easy-cp.yml [hunk 2]
```diff
@@ -29,15 +29,9 @@ jobs:
       - name: Get Cherry Pick PR
         run: |
           echo "COMMIT_SHA=$(echo ${{ github.event.pull_request.merge_commit_sha }})" >> $GITHUB_ENV
-      - name: Checkout Flutter Repo
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
-        with:
-          repository: flutteractionsbot/flutter
-          token: ${{ secrets.FLUTTERACTIONSBOT_CP_TOKEN }}
```
Comment (CaseyHillers) [inline .github/workflows/easy-cp.yml:36]: If the issue is with the token, can we remove the token here? This step shouldn't need any secrets to run as the repos are public.

File: .github/workflows/easy-cp.yml (hunk 1 above)
Comment (XilaiZhang) [reply .github/workflows/easy-cp.yml:55]: nope. it outputs ***@github.com where the raw string is replaced with asterisks. [example workflow](https://github.com/XilaiZhang/miscellaneous-side-project/actions/runs/7548409479/job/20550397014)

File: .github/workflows/easy-cp.yml (hunk 2 above)
Comment (XilaiZhang) [reply .github/workflows/easy-cp.yml:36]: yeah this part is deleted in this PR.

The token is used in the step below where `git push` requires it. The token is required for push if we don't want to set up ssh keys.

Comment (CaseyHillers) [reply .github/workflows/easy-cp.yml:36]: What I'm saying is revert this diff, and remove the token parameter to the checkout action. I'm not sure I understand why we need to run `git clone` locally here as there may be other flakes that surface.

**Review by CaseyHillers — APPROVED** [review]

File: .github/workflows/easy-cp.yml (hunk 2 above)
Comment (XilaiZhang) [reply .github/workflows/easy-cp.yml:36]: Thanks for clarifying! in order to revert the diff, I did more research on why actions/checkout have token/credential issue, and my current understanding is that the `persist-credentials: false` field is needed (added this field in the most recent patch). This ensures that the checkout step uses default token, but does not override the auth token used in the `git push` step below.

Yeah previously I removed actions/checkout entirely to remove its influence on credentials. The most recent patch adds it back.

**Pull request merged**